src/DotNetProjectFile.Analyzers/Analyzers/Slnx/UseSlnxFiles.cs

```diff
@@ -0,0 +1,40 @@
+namespace DotNetProjectFile.Analyzers.Slnx;
+
+/// <summary>Implements <see cref="Rule.SLNX.UseSlnxFiles"/>.</summary>
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class UseSlnxFiles() : MsBuildProjectFileAnalyzer(Rule.SLNX.UseSlnxFiles)
+{
+    /// <inheritdoc />
+    public override IReadOnlyCollection<ProjectFileType> ApplicableTo => ProjectFileTypes.SDK;
+
+    /// <inheritdoc />
+    public override bool DisableOnFailingImport => false;
+
+    /// <inheritdoc />
+    protected override void Register(ProjectFileAnalysisContext context)
+    {
+        if (context.Options.AdditionalFiles.None(f => IOFile.Parse(f.Path).Extension.IsMatch(".slnx"))
+            && !Net90OrUp(context))
+        {
+            context.ReportDiagnostic(Descriptor, context.File);
+        }
+    }
+
+    private static bool Net90OrUp(ProjectFileAnalysisContext context)
```

Comment:
Technically you don't need to be targeting 9.0 to support slnx. You just need a new enough visual studio version or have the .NET 9 sdk installed. I would still like to be recommended to use slnx when working on a .net standard project for example. Wouldn't it be better to just always recommend this and add a "when to disable" note in the docs saying you can disable this rule if you can't switch to slnx due to not being able to install modern tooling?

Reply:
Either Sonar or the NETAnalyzers (or both) also do this if I'm not mistaken.